lib/ClangImporter/ImportMacro.cpp

```diff
@@ -334,6 +345,29 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
         tokenI[3].is(clang::tok::r_paren)) {
       return importStringLiteral(impl, DC, macro, name, tokenI[2],
                                  MappedStringLiteralKind::CFString, ClangN);
+    // Check for literals with a cast, e.g. "(cpu_type_t) 1"
+    } else if (tokenI[0].is(clang::tok::l_paren) &&
+               tokenI[1].is(clang::tok::identifier) &&
```

Comment:
Right, this heuristic is not complete, but covers a few common cases. If we were to include all possible combinations of keywords, it would become quite a bit more complex (`unsigned long long`, `signed char`, `char signed`, …).